bims/api_views/get_biorecord.py

```diff
@@ -157,10 +158,66 @@ def get(self, request):
         collection_with_references = collection_results.exclude(
             source_reference__isnull=True
         ).distinct('source_reference')
-        source_references = [
-            col.source_reference.link_template() for col in
-            collection_with_references
-        ]
+        source_references = []
+        for col in collection_with_references:
```

Comment:
It seems you have redundancy here, there are three codes with the same function. Maybe it's better to put it on the Model Manager, since the logic revolves around collection_record. Here's the example : 

Create Manager and QuerySet in biological_collection_record, 
here I want to get validated count from the queryset result :

```
class BiologicalCollectionQuerySet(models.QuerySet):
    def validated_count(self):
        return self.filter(validated=True).count()


class BiologicalCollectionManager(models.Manager):
    def get_queryset(self):
        return BiologicalCollectionQuerySet(self.model, using=self._db)

    def validated_count(self):
        return self.get_queryset().validated_count()
```

Lastly add the manager to the model : 
```
class BiologicalCollectionRecord(AbstractValidation):
    """Biological collection model."""
    objects = BiologicalCollectionManager()
```

So to get the validated_count, I only need to do this : 
```
bio = BiologicalCollectionRecord.objects.filter(original_species_name__icontains='dae').validated_count()
```

For your case : `references = collection_results.source_references()`